hoomd/mpcd/RejectionVirtualParticleFiller.h

```diff
@@ -0,0 +1,152 @@
+    const BoxDim& box = m_pdata->getBox();
+    const Scalar3 lo = box.getLo();
+    const Scalar3 hi = box.getHi();
+    const unsigned int m_Lx = hi.x - lo.x;
+    const unsigned int m_Ly = hi.y - lo.y;
+    const unsigned int m_Lz = hi.z - lo.z;
+    const unsigned int m_NVirtMax = m_density*(m_Lx*m_Ly*m_Lz);
+
+    // Allocate memory
+    // first remove any previously added memory for virtual particles
+    // This is probably being done already, so that needs to be checked.
+    m_mpcd_pdata->removeVirtualParticles();
+    // Add N virtual particles as a worst case estimate
+    m_mpcd_pdata->addVirtualParticles(m_NVirtMax);
+
+    // Draw particles
+    ArrayHandle<Scalar4> h_pos(m_mpcd_pdata->getPositions(), access_location::host, access_mode::readwrite);
+    ArrayHandle<Scalar4> h_vel(m_mpcd_pdata->getVelocities(), access_location::host, access_mode::readwrite);
+    ArrayHandle<unsigned int> h_tag(m_mpcd_pdata->getTags(), access_location::host, access_mode::readwrite);
+
+    const Scalar vel_factor = fast::sqrt(m_T->getValue(timestep) / m_mpcd_pdata->getMass());
+
+    // index to start filling from
+    unsigned int pidx = m_mpcd_pdata->getN();
+    for (unsigned int i=0; i < m_NVirtMax; ++i)
+        {
+        // TODO: Currently just using the constant for SlitGeometryFiller which needs to be changed.
+        hoomd::RandomGenerator rng(hoomd::RNGIdentifier::SlitGeometryFiller, m_seed, timestep);
+
+        Scalar3 tmp_pos = make_scalar3(hoomd::UniformDistribution<Scalar>(lo.x, hi.x)(rng),
+                                       hoomd::UniformDistribution<Scalar>(lo.y, hi.y)(rng),
+                                       hoomd::UniformDistribution<Scalar>(lo.z, hi.z)(rng));
+
+        if (m_geom->isOutside(tmp_pos))
+            {
+            h_pos.data[pidx] = make_scalar4(tmp_pos.x,
+                                            tmp_pos.y,
+                                            tmp_pos.z,
+                                            __int_as_scalar(m_type));
+
+            hoomd::NormalDistribution<Scalar> gen(vel_factor, 0.0);
+            Scalar3 vel;
+            gen(vel.x, vel.y, rng);
+            vel.z = gen(rng);
+            h_vel.data[pidx] = make_scalar4(vel.x,
+                                            vel.y,
+                                            vel.z,
+                                            __int_as_scalar(mpcd::detail::NO_CELL));
+
+            pidx += 1;
+            }
+        }
+    m_mpcd_pdata->setVirtualParticleNumber(pidx - m_mpcd_pdata->getN());
+    }
+
+namespace detail
+{
+//! Export RejectionVirtualParticleFiller to python
+template<class Geometry>
+void export_RejectionVirtualParticleFiller(pybind11::module& m)
+    {
+    namespace py = pybind11;
+    py::class_<mpcd::RejectionVirtualParticleFiller<Geometry>, std::shared_ptr<mpcd::RejectionVirtualParticleFiller<Geometry>>>
+        (m, "RejectionFiller", py::base<mpcd::VirtualParticleFiller>())
```

Comment:
This needs to include the name of the geometry in the exported class name, or you will get a bunch of symbol clashes.

Reply:
Does it mean that I need to have separate export functions for all the geometries?
I don't really know what I need to do.